### PR TITLE
thumbs fix

### LINKED ISSUE #5166: Redraw lighttable probably missing after import images from card or camera
1. Import new images from card or camera
2. When import is over, see new film roll created (or see already existing film roll if images imported on existing one) and number of images displayed and seems selected but no images displayed on lighttable main view (only the grey background is shown).

I have imported 4 times images this way since a week and all times I reproduce this (minor) issue.

=> Images should be displayed just after import.

=> Clicking after that on the collection and see images displayed. So probably just a missing redraw after importing images from SD card or camera. @AlicVB: maybe for you?

I rarely use import image or from folder ones so not test them.

Tested on last darktable master on Ubuntu 20.04.

### PR DESCRIPTION
**Pull request by AlicVB**
- fix filmstrip rewind in some situation, as mentioned by @esq4 in #5042 
- fix the focus when no mouse movement (key navigation) in culling/preview
- allow generate-cache to work with mipmap 8 (mentioned by @Nilvus )
- refresh view after camera import (fix #5166 )

### CODE REVIEW

File: src/generate-cache/main.c [hunk 1]
```diff
@@ -174,12 +173,12 @@ int main(int argc, char *arg[])
     else if((!strcmp(arg[k], "-m") || !strcmp(arg[k], "--max-mip")) && argc > k + 1)
     {
       k++;
-      max_mip = (dt_mipmap_size_t)MIN(MAX(atoi(arg[k]), 0), 7);
+      max_mip = (dt_mipmap_size_t)MIN(MAX(atoi(arg[k]), 0), 8);
```
Comment (TurboGit) [inline src/generate-cache/main.c:176]: Instead of 0 and 8, why not sure DT_MIPMAP_0 and DT_MIPMAP_8 ?

File: src/generate-cache/main.c [hunk 2]
```diff
@@ -174,12 +173,12 @@ int main(int argc, char *arg[])
...
     }
     else if(!strcmp(arg[k], "--min-mip") && argc > k + 1)
     {
       k++;
-      min_mip = (dt_mipmap_size_t)MIN(MAX(atoi(arg[k]), 0), 7);
+      min_mip = (dt_mipmap_size_t)MIN(MAX(atoi(arg[k]), 0), 8);
```
Comment (TurboGit) [inline src/generate-cache/main.c:181]: Likewise

File: src/generate-cache/main.c (hunk 1 above)
Comment (AlicVB) [reply src/generate-cache/main.c]: you're right of course. I've just keeped the previous notation, but using DT_MIPMPA8? is more clear ! done

File: src/generate-cache/main.c (hunk 2 above)
Comment (AlicVB) [reply src/generate-cache/main.c]: done

**Comment by TurboGit** [comment]
Looks good now, merging. Thanks.

**Comment by AxelG-DE** [comment]
@AlicVB 
that "generate-cache" is that meant to be called like this?
`/darktable-generate-cache -m 8`

If **no**, I will shut up :)

If **yes**, it once segfaulted (build="Release", sorry) and two times stopped with 
`munmap_chunk(): invalid pointer`

[darktable_bt_XUISK0.txt](https://github.com/darktable-org/darktable/files/4675371/darktable_bt_XUISK0.txt)

**Comment by Nilvus** [comment]
@AxelG-DE: I often use this script and never had issues with it. Even with that update and -m 8 option (it runs actually on my system since yesterday evening).

It's yes darktable-generate-cache -m 8 (I assume that / you add on start is or ./ if you are in the script folder or full path).

Not sure but as you seems to have strange issue actually, maybe something related to your system (or your config folder) more than darktable himself...

**Comment by AlicVB** [comment]
@AxelG-DE : yes, it can be called like that. I don't think that your issue is tied with the new "8" limit, as the process to generate mipmap 7 or 8 is the same afaik.
Sadly (but you know that) the backtrace isn't helpful, neither your errors...
Can you please open a new issue with backtrace with debugging symbol if you reproduce ? Thanks

**Comment by Nilvus** [comment]
@AxelG-DE: I don't read all issues so don't expect me to know all. I have just proposed that at a hypothesis, nothing more. And propose all ideas, even the less probable ones is to take seriously an issue, not the opposite. My first job was network administrator (not sure if this is the good english term but so manage a computer park, solve issues on it and update it ; and on most of the main professionals OS (even some Unix ones) with quite big companies behind this enterprise as clients) and be sure that I found sometimes some quite really strange ones (where the strange thing was not the issue but the reason of the issue behind).

So said that, even if the script is supposed to check that, do you have the 100% thumbnails option checked on preferences? Please, as @AlicVB propose, do not answer here and create a new issue to post debugging output and answers here.

**Comment by AxelG-DE** [comment]
@Nilvus,
didn't want to offend you either and also did not mean you wouldn't take things serious :)

sorry keep on talking here, as I am not ready for BR, yet. 
- tried initial status and still failed
- Recompiled w/o CFLAGS and with RelWithDebInfo issue gone =:-/
- Recompiled again with CFLAGS and Release, issue gone =:-/
(don't say a word, I know what you wanna say :-> ) ...is still running now on a fresh ~/.cache/darktable

> do you have the 100% thumbnails option checked on preferences?

I was looking around in prefs and do not get this question right. Where should I check wich option 

grafik

**Comment by Nilvus** [comment]
> @Nilvus,
> didn't want to offend you either and also did not mean you wouldn't take things serious :)

Don't worry about that and you offend nobody. It's all good and just explanation.

>     * Recompiled again with CFLAGS and Release, issue gone =:-/
>       (don't say a word, I know what you wanna say :-> ) ...is still running now on a fresh ~/.cache/darktable

Not here...

> 
> > do you have the 100% thumbnails option checked on preferences?
> 
> I was looking around in prefs and do not get this question right. Where should I check wich option

On cpu/gpu/memory section, check 3rd line.

**Comment by AxelG-DE** [comment]
> Not here...

:-)



> On cpu/gpu/memory section, check 3rd line.

Checked, is on.

I'll keep silent here now and BR if issue comes back :)

**Comment by AxelG-DE** [comment]
Hey guys,

just want to inform, there will no bug report come...
after last conversation I made it several times to crash but always with empty backtrace, even it shouldn't (no cflags, build RelWithDebInfo).

With the help of @parafin on IRC we checked whether my machine in general can provide debug symbols, -it can. 

In total >75h running inside (gdb) 1x used "make", 1x used "ninja". No crash, no output. Today last run not within gdb, no crash either. 

So I can only say, sorry for the noise and thanks for your patience.
Cheers
 Axel

**Pull request merged**